.github/workflows/codeql.yml

```
@@ -0,0 +1,33 @@
+name: "CodeQL"
+
+on:
+  pull_request:
+    branches: [ "master" ]
```

nit: wdyt about adding some paths to ignore so that in case of simple change PRs the workflow is not triggered
```suggestion
    branches: [ "master" ]
    paths-ignore:
      - '**/*.md'
```